galaxy_ng/app/api/v3/serializers/namespace.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import re
+
+from django.db import transaction
+
+from rest_framework.exceptions import ValidationError
+from rest_framework import serializers
+
+from galaxy_ng.app import models
+from galaxy_ng.app.models import auth as auth_models
+from galaxy_ng.app.api.v3.serializers.group import GroupSummarySerializer
+
+log = logging.getLogger(__name__)
+
+
+class NamespaceLinkSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.NamespaceLink
+        fields = ('name', 'url')
+
+
+class NamespaceSerializer(serializers.ModelSerializer):
+    links = NamespaceLinkSerializer(many=True, required=False)
+
+    groups = GroupSummarySerializer(many=True)
+
+    class Meta:
+        model = models.Namespace
+        fields = (
+            'id',
+            'name',
+            'company',
+            'email',
+            'avatar_url',
+            'description',
+            'links',
+            'groups',
+            'resources'
+        )
+
+    # replace with a NamespaceNameSerializer and validate_name() ?
+    def validate_name(self, name):
+        if not name:
+            raise ValidationError(detail={
+                'name': "Attribute 'name' is required"})
+        if not re.match(r'^[a-zA-Z0-9_]+$', name):
+            raise ValidationError(detail={
+                'name': 'Name can only contain [A-Za-z0-9_]'})
+        if len(name) <= 2:
+            raise ValidationError(detail={
+                'name': 'Name must be longer than 2 characters'})
+        if name.startswith('_'):
+            raise ValidationError(detail={
+                'name': "Name cannot begin with '_'"})
+        return name
+
+    def create(self, validated_data):
```

Comment:
Shouldn't namespace creation be atomic?

Reply:
Hmm, I have no idea. 

I assume ATOMIC_REQUEST=True, so viewset actions should be automatically wrapped in transactions.

For serializers though... I don't know. DRF docs don't mention it at all. Django docs never seem to mention it in context of serializers. 

The create() currently doesn't do any try/except, so it should be okay to @atomic.transaction (based on 
the exception warning in https://docs.djangoproject.com/en/3.0/topics/db/transactions/#django.db.transaction.atomic)

Reply:
Seems to with it, so I'll add it.

Reply:
By default in Django `ATOMIC_REQUEST=False`. Also I wouldn't recommend relying on a setting parameter.